src/services/formatting/formatting.ts

```diff
@@ -1152,6 +1152,48 @@ namespace ts.formatting {
         }
     }
 
+    /**
+     * Gets the indentation level of the multi-line comment enclosing position,
+     * and a negative value if the position is not in a multi-line comment.
+     */
+    export function getIndentationOfEnclosingMultiLineComment(sourceFile: SourceFile, position: number, options: EditorSettings): number {
+        const range = getRangeOfEnclosingComment(sourceFile, position, /*onlyMultiLine*/ true);
+        if (range) {
+            const commentStart = range.pos;
+            const commentLineStart = getLineStartPositionForPosition(commentStart, sourceFile);
+            const { column, character } = SmartIndenter.findFirstNonWhitespaceCharacterAndColumn(commentLineStart, commentStart, sourceFile, options);
+            return column + /*length after whitespace ends*/ range.pos - (commentLineStart + character);
+        }
+        return undefined;
+    }
+
+    export function getRangeOfEnclosingComment(sourceFile: SourceFile, position: number, onlyMultiLine: boolean): CommentRange | undefined {
+        // Considering a fixed position,
+        // - trailing comments are those following and on the same line as the position.
+        // - leading comments are those in the range [position, start of next non-trivia token)
+        // that are not trailing comments of that position.
+        //
+        // Note, `node.start` is the start-position of the first comment following the previous
+        // token that is not a trailing comment, so the leading and trailing comments of all
+        // tokens contain all comments in a sourcefile disjointly.
+        const precedingToken = findPrecedingToken(position, sourceFile);
+        const trailingRangesOfPreviousToken = precedingToken && getTrailingCommentRanges(sourceFile.text, precedingToken.end);
+        const leadingCommentRangesOfNextToken = getLeadingCommentRangesOfNode(getTokenAtPosition(sourceFile, position, /*includeJsDocComment*/ false), sourceFile);

```

Comment:
What is the impact of `/*includeJsDocComment*/ false`?  Is there a test that depends on the value?